cluster/crds/pkg.crossplane.io_functionrevisions.yaml

```diff
@@ -333,3 +333,319 @@ spec:
+                  controller needs these permissions to run. The RBAC manager is
+                  responsible for granting them.
+                items:
+                  description: |-
+                    PolicyRule holds information that describes a policy rule, but does not contain information
+                    about who the rule applies to or which namespace the rule applies to.
+                  properties:
+                    apiGroups:
+                      description: |-
+                        APIGroups is the name of the APIGroup that contains the resources.  If multiple API groups are specified, any action requested against one of
+                        the enumerated resources in any API group will be allowed. "" represents the core API group and "*" represents all API groups.
+                      items:
+                        type: string
+                      type: array
+                      x-kubernetes-list-type: atomic
+                    nonResourceURLs:
+                      description: |-
+                        NonResourceURLs is a set of partial urls that a user should have access to.  *s are allowed, but only as the full, final step in the path
+                        Since non-resource URLs are not namespaced, this field is only applicable for ClusterRoles referenced from a ClusterRoleBinding.
+                        Rules can either apply to API resources (such as "pods" or "secrets") or non-resource URL paths (such as "/api"),  but not both.
+                      items:
+                        type: string
+                      type: array
+                      x-kubernetes-list-type: atomic
+                    resourceNames:
+                      description: ResourceNames is an optional white list of names
+                        that the rule applies to.  An empty set means that everything
+                        is allowed.
+                      items:
+                        type: string
+                      type: array
+                      x-kubernetes-list-type: atomic
+                    resources:
+                      description: Resources is a list of resources this rule applies
+                        to. '*' represents all resources.
+                      items:
+                        type: string
+                      type: array
+                      x-kubernetes-list-type: atomic
+                    verbs:
+                      description: Verbs is a list of Verbs that apply to ALL the
+                        ResourceKinds contained in this rule. '*' represents all verbs.
+                      items:
+                        type: string
+                      type: array
+                      x-kubernetes-list-type: atomic
+                  required:
+                  - verbs
+                  type: object
+                type: array
+            type: object
+        type: object
+    served: true
+    storage: false
```

Comment:
So, it seems you're updating the storage configuration from v1beta1 to v1, right? As a next step, before we completely remove v1beta1, we'll need to perform a storage migration similar to what we're doing in the initialization process https://github.com/crossplane/crossplane/blob/master/cmd/crossplane/core/init.go#L75-L76 I want to include this here because we've issues with missing storage migration for locks and composition revisions in the past

Reply:
Right now we don't have any plans to remove v1beta1. We can open a tracking issue to do that, and make note of this there. I think we should keep it for at least 3 releases - so we'd drop it no sooner than v1.20.

Reply:
Oh, I missed that we already had code to do the migration. I can add that in this PR.

Reply:
Done.